set array search_builder_type for enumField

hypy13 opened the pull request.

nimaxin commented:
I don’t believe the issue has been fully resolved yet.
![Capture2](https://github.com/user-attachments/assets/5b0a4bbd-2489-449d-8a07-d4b4f49668ac)

The functionality should work like the "Office" or "Position" fields in [this DataTables SearchBuilder example](https://datatables.net/extensions/searchbuilder/examples/performance/searchBuilder5k.html).
![expected-result](https://github.com/user-attachments/assets/9a02f78d-4664-4803-b050-942ee4974d0f)

hypy13 commented:
hi, can you merge it
then we fix it for next step

hypy13 commented:
@nimaxin 

Hey! I've fixed the issue with array filters by creating a new module called search_builder_conditions.js. This allows us to define custom condition functions for different field types, like boolean, default, and now array . 

In this update, I added support for array filtering using the = and != operators. These conditions are now configurable in the searchBuilder settings like this: 

list.js file , Line 206
```
array: {
  '=': SearchBuilderConditions.array.equal(model.fields),
  '!=': SearchBuilderConditions.array.notEqual(model.fields),
}
```

This setup makes it easier to extend or override other conditions later if needed. 

Let me know what you think!
